src/test/java/picard/sam/FastqToSamTest.java

```diff
@@ -177,6 +181,67 @@ public void testEmptyFastqAllowed() throws IOException {
         convertFile(emptyFastq, null, FastqQualityFormat.Illumina, false, false, true);
     }
 
+    @Test
+    public void testStreamInputWithoutQuality() throws IOException {
+        ByteArrayOutputStream stderrStream = new ByteArrayOutputStream();
+        PrintStream newStderr = new PrintStream(stderrStream);
+        PrintStream oldStderr = System.err;
+        System.setErr(newStderr);
+
+        final File tmpFile = File.createTempFile("empty", ".sam");
+
+        final String[] args = {
+                "FASTQ=/dev/stdin",
+                "SAMPLE_NAME=sample001",
+                "OUTPUT=" + tmpFile
+        };
+
+        Assert.assertEquals(runPicardCommandLine(args), 1);
+        System.setErr(oldStderr);
+        Assert.assertTrue(stderrStream.toString().endsWith("QUALITY_FORMAT must be specified when input is not a regular file\n"));
+    }
+
+    @Test
+    public void testStreamInput() throws IOException {
+        final File output = newTempSamFile("stdin");
+        String[] command = {
+                "/bin/bash",
+                "-c",
+                "echo -ne '" +
+                "@ERR194147.10008417/1\\n" +
+                "ATTTAATTAAGAAAATGTAAACTAAATGACAGTAGACAGACAAGTATGCCTTTGC\\n" +
+                "+\\n" +
+                "???????????????????????????????????????????????????????\\n'" +
+                "|" +
+                "java -classpath " +
+                        CLASSPATH +
+                        "picard.cmdline.PicardCommandLine " +
+                        "FastqToSam " +
+                        "-FASTQ /dev/stdin " +
+                        "-QUALITY_FORMAT Standard " +
+                        "-SAMPLE_NAME sample001 " +
+                        "-OUTPUT " + output 
+        };
+
+        try {
+            ProcessBuilder processBuilder = new ProcessBuilder(command);
+            processBuilder.inheritIO();
+            Process process = processBuilder.start();
+            Assert.assertEquals(process.waitFor(), 0);
+        } catch (Exception e) {
+            Assert.fail("Failed to pipe data from stdin to FastqToSam", e);
+        }
+        final SamReader samReader = SamReaderFactory.makeDefault().open(output);
```

Comment:
I think you can save a few lines and handle closing things more cleanly with something like this
```
        try(final SamReader samReader = SamReaderFactory.makeDefault().open(output)) {
            long actualCount = samReader.iterator().stream().count();
            Assert.assertEquals(actualCount, 1);
        }
```

Reply:
Done.